protos/flyteidl/core/tasks.proto

```diff
@@ -154,4 +154,44 @@ message Container {
     // Ports to open in the container. This feature is not supported by all execution engines. (e.g. supported on K8s but
     // not supported on AWS Batch)
     repeated ContainerPort ports = 7;
+
+    // BETA: This enables use of CoPilot. This makes it possible to to run a completely portable container, that uses inputs and outputs
+    // only from the local file-system and without having any reference to flyteidl. This is supported only on K8s at the moment.
+    // If CoPilot is enabled, then data will be mounted in accompanying directories specified in the CoPilot settings. If the directories 
+    // are not specified, inputs will be mounted onto and outputs will be uploaded from a pre-determined file-system path. Refer to the documentation
+    // to understand the default paths.
+    bool use_copilot = 8;
+
+    // Optional configuration for CoPilot. If not specified, then default values are used.
+    CoPilot copilot_config = 9;
+}
+
+// This configuration allows executing raw containers in Flyte using the Flyte CoPilot system.
+// Flyte CoPilot, eliminates the needs of flytekit or sdk inside the container. Any inputs required by the users container are side-loaded in the input_path
+// Any outputs generated by the user container - within output_path are automatically uploaded.
+message CoPilot {
+  // MetadataFormat decides the encoding format in which the input metadata should be made available to the containers. 
+  // If the user has access to the protocol buffer definitions, it is recommended to use the PROTO format.
+  // JSON and YAML do not need any protobuf definitions to read it
+  // All remote references in core.LiteralMap are replaced with local filesystem references (the data is downloaded to local filesystem)
+  enum MetadataFormat {
+    // JSON / YAML are serialized represnetation of a map[string]primitive. primitive -> int, string, bool, double/float, bytes
+    JSON = 0;
```

Comment:
Can you add a sample JSON structure to the docs here...

Reply:
true, will do